### PR TITLE
[MRG+1] Require sklearn >= 0.18

### LINKED ISSUE #4788: Is scikit-learn required for ICA?
When creating an `ICA` object, I get the following error if scikit-learn is not installed:

    RuntimeError: the scikit-learn package (version >= 0.12)is required for ICA

I think MNE's ICA implementation uses the `FastICA` class from scikit-learn, but AFAIK it ships its own (Extended) Infomax algorithm, in which case scikit-learn should not be required.

### PR DESCRIPTION
**Pull request by cbrnr**
Fix #4788

### CODE REVIEW

**Comment by cbrnr** [comment]
There's actually one Travis job which has sklearn=0.14.

**Comment by larsoner** [comment]
I guess our Readme might be incorrect. @agramfort what minimum `sklearn` should we require?

**Comment by cbrnr** [comment]
`mne.preprocessing.ica` lists [`sklearn` >= 0.12](https://github.com/cbrnr/mne-python/blob/187f098409b9688c91f7843bdafec97db2df9aa5/mne/preprocessing/ica.py#L246). There are a ton of tests that require 0.17, and some tests that require 0.15, and one test that requires 0.18 (but they just might not run if a lower version is installed). `mne.cov.compute_covariance` requires [`sklearn` >= 0.15](https://github.com/cbrnr/mne-python/blob/086478e8334b38fbe02be4c0f2a147429906ba4a/mne/cov.py#L747).

A good minimum version would be 0.15 if we are OK with some tests not running. Otherwise, it should be 0.18.

**Comment by agramfort** [comment]
I am fine with 0.15 but we should have 2 builds one that runs 0.15 and one than runs 0.18.

**Comment by cbrnr** [comment]
By builds you mean Travis CI builds?

**Comment by larsoner** [comment]
IIRC we currently have one that runs 0.14 and the others run whatever the latest is (0.19.1?), do you want to change one explicitly to run 0.18 (and bump the 0.14 to 0.15)?

**Comment by agramfort** [comment]
ok then let's make 0.15 on one build and then latest conda package as it is
now

**Comment by cbrnr** [comment]
And in the docs we say minimum sklearn >= 0.15

**Comment by cbrnr** [comment]
Done, let me know if this is all that's needed to change the Travis builds.

File: README.rst [hunk 1]
```diff
@@ -79,7 +79,7 @@ The minimum required dependencies to run MNE-Python are:
 
 For full functionality, some functions require:
 
-- Scikit-learn >= 0.18
+- Scikit-learn >= 0.15
```
Comment (agramfort) [inline README.rst]: say that 0.18 is recommended

**Comment by cbrnr** [comment]
Done

**Comment by larsoner** [comment]
Failure is due to `kernel_ridge` appearing in 0.16, not 0.15 -- @cbrnr can you update the decorator for that test?

**Comment by cbrnr** [comment]
Done.

**Comment by larsoner** [comment]
Thanks @cbrnr

**Pull request merged**